### ISSUE TITLE
[Enhancement]: Warn against using from_port and to_port set to 0 with non-all protocol

### ISSUE THREAD

**Issue by khumps** [hashicorp/terraform-provider-aws#27525]
### Description

```
resource "aws_security_group" "eks-ingress" {
  vpc_id      = module.vpc.vpc_id
  name_prefix = "${var.stratum} eks-ingress"
  ingress {
      from_port = 0
      to_port = 0
      protocol = "-1"
      cidr_blocks = ["192.168.1.1/32"]
  }
}
```
Creates the following rule as expected
![image](https://user-images.githubusercontent.com/7849025/198416931-91166c69-5aba-440e-894c-53d3f8b89c67.png)

I would expect
```
resource "aws_security_group" "eks-ingress" {
  vpc_id      = module.vpc.vpc_id
  name_prefix = "${var.stratum} eks-ingress"
  ingress {
      from_port = 0
      to_port = 0
      protocol = "tcp"
      cidr_blocks = ["192.168.1.1/32"]
  }
}
```
to create almost the same rule (all ports, only TCP) but it creates the rule with only the port 0. 
![image](https://user-images.githubusercontent.com/7849025/198417491-38a0806f-0864-4770-9b66-c75f3b297605.png)

Looking at the code, I am guessing that this is an AWS API issue, not a bug in the provider. However, we could probably have a similar warning preventing someone using from_port and to_port both equal to zero as the provider has a warning for not doing that if using all protocols.

### Affected Resource(s) and/or Data Source(s)

* aws_security_group
* aws_security_group_rule

### Potential Terraform Configuration

```terraform
if aws.StringValue(perm.IpProtocol) != "-1" && aws.Int64Value(perm.ToPort) == 0) {
			return nil, fmt.Errorf(
				"you must specify port 65535 as the to_port if you want all ports on a specific protocol")
		}
```

### Would you like to implement a fix?

Yes

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/issues/27079.

**Comment by foges** [comment]
Yes, please add a warning about this! It took me a day to figure out that this was the reason my instance wasn't reachable
